### PR TITLE
Failing test for first class collections

### PR DESCRIPTION
**Pull request by Majkl578**
Failing test for #942 - empty first class collections serialized as object.

### CODE REVIEW

File: tests/Fixtures/FirstClassCollection.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+namespace JMS\Serializer\Tests\Fixtures;
+
+use JMS\Serializer\Annotation as Serializer;
+
+class FirstClassCollection implements \IteratorAggregate
+{
+    /**
+     * @Serializer\Type("array<int>")
+     * @Serializer\Inline
```
Comment (goetas) [inline tests/Fixtures/FirstClassCollection.php]: Currently, your first test case works just by coincidence. `@Inline` was never tested on collections.
If will also fail if your you unset one item from the collection (making keys not consequent).

And on top of it what it the expected result if you add one more property to the  `FirstClassCollection` ?

Comment (Majkl578) [reply tests/Fixtures/FirstClassCollection.php]: > what it the expected result if you add one more property

Depends. Either an error as there could only be one property with ``@Inline, or, if you consider things like exclusion strategy, error of there is more than one property after applying exclusion rules.

**Comment by goetas** [comment]
However could be a nice feature to have. So PR are welcome

File: tests/Fixtures/FirstClassCollection.php (hunk 1 above)
Comment (goetas) [reply tests/Fixtures/FirstClassCollection.php]: An error (doable at "metadata" discovery time) should be fine

Comment (goetas) [reply tests/Fixtures/FirstClassCollection.php]: Consider that the collection could be also an associative array, so it can be both `[]` and `{}`

**Comment by goetas** [comment]
fixed in https://github.com/schmittjoh/serializer/pull/944

**Pull request closed without merging**